### PR TITLE
Add support for --userns=nomap

### PR DESCRIPTION
**Pull request by rhatdan**
From a security point of view, it would be nice to be able to map a
rootless usernamespace that does not use your own UID within the
container.

This would add protection against a hostile process escapping the
container and reading content in your homedir.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: pkg/util/utils.go
Comment (Luap99) [inline pkg/util/utils.go:424]: Why should this be a noop for root? IMO it would make sense to use this as root when you want to make sure host uid 0 is not mapped into the container.
If this does not make sense for some reason it should error instead.

Comment (rhatdan) [reply pkg/util/utils.go:424]: For root you should use --userns=auto, which does the same thing, without allocating the entire namespace of UIDs.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: pkg/util/utils.go
Comment (Luap99) [reply pkg/util/utils.go:424]: If you think so it should error, otherwise we are just ignoring user input.

**Review by Luap99 — COMMENTED** [review]
I think you also have to update the podman pod create man page.

File: pkg/util/utils.go
Comment (rhatdan) [reply pkg/util/utils.go:424]: We are also currently ignoring the --userns=keep-id.

File: pkg/util/utils.go
Comment (giuseppe) [inline pkg/util/utils.go:406]: I think we should raise an error if `len(uids) == 0` or `len(gids) == 0`, since there are no additional IDs that are usable for the current user.

Comment (giuseppe) [reply pkg/util/utils.go:406]: currently it fails later on but the error is not clear: `Error: must provide at least one UID or GID mapping to configure a user namespace`

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by eriksjolund** [comment]
I think	both `sudo podman run --userns=keep-id__ ...` and	`sudo podman run --userns=nomap ...` should raise errors
as it would cause less confusion among the users. An error text could also guide the user in the right
direction. Also there would be no need to reply to support questions with
_Did you use the --userns=nomap as root	or rootless?_

File: docs/source/markdown/podman-create.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:1227]:
```suggestion
keep-id   |$UID           |$UID (Map user account to same UID within container.)
```

File: docs/source/markdown/podman-create.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:1229]: You have an ending period '.' for keep-id within the parens, which I prefer.  Your pick, but you should keep them consistent.

File: docs/source/markdown/podman-pod-create.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-pod-create.1.md:319]: Same comments as above.

File: docs/source/markdown/podman-run.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:1297]: ditto comments from prior.

**Comment by TomSweeneyRedHat** [comment]
All kinds of test unhappiness @rhatdan

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

Extra :cookie: for the tests improvements!

File: test/system/500-networking.bats
Comment (Luap99) [inline test/system/500-networking.bats:91]: Can you change the keep-id for some other namespace that works both with root and rootless? I guess a custom --uidmap should work? I think it is important to run this test as root as well.

Comment (rhatdan) [reply test/system/500-networking.bats:91]: Let's open a PR to add a test for --userns=auto.

**Pull request merged**
